Swift/BRCrypto/BRCryptoSystem.swift

```diff
@@ -496,110 +499,194 @@ public final class System {
+            let blockchainsGroup = DispatchGroup ()
+
+            // query blockchains
+            self.query.getBlockchains (mainnet: self.onMainnet) { (blockchainResult: Result<[BlockChainDB.Model.Blockchain],BlockChainDB.QueryError>) in
+                // Filter our defaults to be `self.onMainnet` and supported (non-nil blockHeight)
+                let blockChainModelsDefaults = System.defaultBlockchains
+                    .filter { $0.isMainnet == self.onMainnet && nil != $0.blockHeight }
+
+                let blockChainModelsRemote = blockchainResult
+                    // Only supported blockchains, but we'll merge in the defaults to handle
+                    // blockchains not supported by BDB.
+                    .map { $0.filter { nil != $0.blockHeight } }
+                    // On error, return [] - we'll use defaults
+                    .getWithRecovery { (ignore) in return [] }
+
+                let blockChainModels =
+                    self.configureMergeBlockchains (builtin: blockChainModelsDefaults,
+                                                    remote: blockChainModelsRemote)
+
+                // If there are no models, then we are done (and disappointed).
+                guard !blockChainModels.isEmpty
+                    else { blockchainsSemaphore.signal(); return }
+
+                // Enter the group once for each model; we'll leave as each currency is processed.
+                // when all have left, self.queue will unblock
+                blockChainModels.forEach { (ignore) in blockchainsGroup.enter() }
+
+                // Signal the dispatch semaphore, the self.queue will now start blocking
+                // on the dispatch group.
+                blockchainsSemaphore.signal()
+
+                // Handle each blockchain Model - there will be at least one model
+                blockChainModels
+                    .forEach { (blockchainModel: BlockChainDB.Model.Blockchain) in
+                        // query currencies
+                        self.query.getCurrencies (blockchainId: blockchainModel.id) { (currencyResult: Result<[BlockChainDB.Model.Currency],BlockChainDB.QueryError>) in
+                            // We get one `currencyResult` per blockchain.  If we leave the group
+                            // upon completion of this block, we'll match the `enter` calls.
+                            defer { blockchainsGroup.leave() }
+
+                            // Find applicable defaults by `blockchainID`
+                            let defaults = System.defaultCurrencies
+                                .filter { $0.blockchainID == blockchainModel.id }
+
+                            // Find applicable application currencies by `blockchainID`
+                            let apps = applicationCurrencies
+                                .filter { $0.blockchainID == blockchainModel.id }
+
+                            // Merge in `defaults` with the result; but, on error, use apps
```

Comment:
Does this give them the behaviour they want? Don't they want apps to always be merged in, otherwise they risk a wallet that the user expects?

Reply:
[Drat 'cancelled my review'; lost my comment here.]

I don't want to really on the currencies provided by the App - the BDB is the definitive source and the App should only be used as a stop-gap for a dead BDB.  That is part of the reason why the interface uses `[BlockChainDB.Model.Currency]` for the argument - it is a low-level thing substituting for a dead BDB.